### PR TITLE
Addition of BigQuery System Tables Reports

### PR DESCRIPTION
**Pull request by neemay**
This PR includes the scripts and documentation for the public BigQuery System Tables Reports dashboard.

### CODE REVIEW

**Comment by jaketf** [comment]
R: @ryanmcdowell 
CC: @danieldeleo

File: views/system_tables/sql/current_assignments.sql [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-- This table retrieves the latest slot capacity for each reservation
+WITH latest_slot_capacity as (
+ SELECT
+   rcp.reservation_name, rcp.slot_capacity
+ FROM
+    `region-{region_name}`.INFORMATION_SCHEMA.RESERVATION_CHANGES_BY_PROJECT AS rcp
+  WHERE
+    -- This subquery returns the latest slot capacity for each reservation
+    -- by extracting the reservation with the maximum timestamp
+    ( rcp.reservation_name, rcp.change_timestamp) IN (
```
Comment (ryanmcdowell) [inline views/system_tables/sql/current_assignments.sql:38]: Remove space before `rcp.reservation_name`

File: views/system_tables/sql/reservation_utilization_month.sql [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  SELECT
+    rcp.reservation_name,
+    rcp.slot_capacity,
+    CONCAT("{project_id}:{location}.", rcp.reservation_name) AS reservation_id,
+  FROM
+    `region-{region_name}`.INFORMATION_SCHEMA.RESERVATION_CHANGES_BY_PROJECT AS rcp
+  WHERE
+    -- This subquery returns the latest slot capacity for each reservation
+    -- by extracting the reservation with the maximum timestamp
+    ( rcp.reservation_name, rcp.change_timestamp) IN (
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_month.sql:56]: Remove space before `rcp.reservation_name`

File: views/system_tables/sql/reservation_utilization_month.sql [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  lsc.slot_capacity AS latest_capacity
+FROM
+  `region-{region_name}`.INFORMATION_SCHEMA.JOBS_BY_ORGANIZATION jbo
+-- Join the slot capacity history
+LEFT JOIN
+  reservation_slot_capacity rsc
+ON
+  jbo.reservation_id = rsc.reservation_id
+  AND jbo.creation_time >= rsc.start_time
+  AND jbo.creation_time < rsc.end_time
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_month.sql:81]: Indent 2 spaces

File: views/system_tables/sql/reservation_utilization_month.sql [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+-- Join the latest slot capacity
+LEFT JOIN
+  latest_slot_capacity lsc
+ON
+  jbo.reservation_id = lsc.reservation_id
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_month.sql:86]: Indent 2 spaces

File: views/system_tables/sql/reservation_utilization_month.sql [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+WHERE
+  -- Includes jobs created 31 days ago but completed 30 days ago
+  jbo.creation_time BETWEEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 31 DAY)
+  AND CURRENT_TIMESTAMP()
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_month.sql:90]: Place `BETWEEN` on new line indented by 2 spaces. Indent `AND` the same

go/sqlstyle#between

File: views/system_tables/sql/reservation_utilization_month.sql [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  AND jbo.end_time BETWEEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 30 DAY)
+  AND CURRENT_TIMESTAMP()
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_month.sql:92]: Same as above

File: views/system_tables/sql/reservation_utilization_week.sql [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+-- This table retrieves the slot capacity history of every reservation
+-- including the start and end time of that capacity
+WITH reservation_slot_capacity AS (
+  SELECT
+    -- Concatenation is needed as RESERVATION_CHANGES_BY_PROJECT only
+    -- includes reservation name but in order to join with
+    -- JOBS_BY_ORGANIZATION, reservation id is required
+    CONCAT("{project_id}:{location}.", reservation_name) AS reservation_id,
+    change_timestamp AS start_time,
+    IFNULL(LEAD(change_timestamp) OVER (PARTITION BY reservation_name ORDER BY change_timestamp ASC),
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_week.sql:39]: Place `LEAD` on the next line

File: views/system_tables/sql/reservation_utilization_week.sql [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  SELECT
+    rcp.reservation_name,
+    rcp.slot_capacity,
+    CONCAT("{project_id}:{location}.", rcp.reservation_name) AS reservation_id,
+  FROM
+    `region-{region_name}`.INFORMATION_SCHEMA.RESERVATION_CHANGES_BY_PROJECT AS rcp
+  WHERE
+    -- This subquery returns the latest slot capacity for each reservation
+    -- by extracting the reservation with the maximum timestamp
+    ( rcp.reservation_name, rcp.change_timestamp) IN (
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_week.sql:56]: Remove space before `rcp.reservation_name`

File: views/system_tables/sql/reservation_utilization_week.sql [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  lsc.slot_capacity AS latest_capacity
+FROM
+  `region-{region_name}`.INFORMATION_SCHEMA.JOBS_BY_ORGANIZATION jbo
+-- Join the slot capacity history
+LEFT JOIN
+  reservation_slot_capacity rsc
+ON
+  jbo.reservation_id = rsc.reservation_id
+  AND jbo.creation_time >= rsc.start_time
+  AND jbo.creation_time < rsc.end_time
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_week.sql:80]: Indent by 2 spaces

File: views/system_tables/sql/reservation_utilization_week.sql [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+-- Join the latest slot capacity
+LEFT JOIN
+  latest_slot_capacity lsc
+ON
+  jbo.reservation_id = lsc.reservation_id
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_week.sql:85]: Indent by 2 spaces

File: views/system_tables/sql/reservation_utilization_week.sql [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+WHERE
+  -- Includes jobs created 8 days ago but completed 7 days ago
+  jbo.creation_time BETWEEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 8 DAY)
+  AND CURRENT_TIMESTAMP()
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_week.sql:89]: Place `BETWEEN` on new line indented by 2 spaces. Indent AND the same

go/sqlstyle#between

File: views/system_tables/sql/reservation_utilization_week.sql [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  AND jbo.end_time BETWEEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 7 DAY)
+  AND CURRENT_TIMESTAMP()
```
Comment (ryanmcdowell) [inline views/system_tables/sql/reservation_utilization_week.sql:91]: Same as above

**Pull request merged**